CRM/Contribute/Form/Contribution.php

```diff
@@ -1453,6 +1457,19 @@ protected function submit($submittedValues, $action, $pledgePaymentID) {
       if (!empty($this->_payNow)) {
         $this->_params['contribution_id'] = $this->_id;
       }
+      // Since we are hiding the payment details block in edit mode,
```

Comment:
This smells bad to me. It's pulling in one value for payment instrument and check number when there may be many associated with a contribution. We should eliminate that expectation from the values submitted when editting a contribution.

Reply:
yep agree - the bao should load them if it needs them. There is a bit of an issue in some of the BAO functions where they are not properly interpretting 'this var is not in $params so we should understand it is not being changed'.

(I'm not going to try to improve this ATM as I think the PR is more seriously blocked at the moment)